File: georocket-server/src/main/java/io/georocket/tasks/ReceivingTask.java
package io.georocket.tasks;

/**
 * A task started by the {@link io.georocket.http.StoreEndpoint} when it
 * receives a file from the client
 * @author Michel Kraemer
 */
public class ReceivingTask extends AbstractTask {
  /**
   * Package-visible default constructor
   */
  ReceivingTask() {
    // nothing to do here
  }

  /**
   * Default constructor
   * @param correlationId the correlation ID this task belongs to
   */
  public ReceivingTask(String correlationId) {
    super(correlationId);
  }
}

File: kotlin_impl/src/main/kotlin/io/georocket/tasks/ReceivingTask.kt
package io.georocket.tasks

import io.georocket.util.UniqueID
import java.time.Instant

/**
 * A task started by the [io.georocket.http.StoreEndpoint] when it
 * receives a file from the client
 * @author Michel Kraemer
 */
data class ReceivingTask(
  override val id: String = UniqueID.next(),
  override val correlationId: String,
  override val startTime: Instant = Instant.now(),
  override val endTime: Instant? = null,
  override val error: TaskError? = null
) : Task
